sbi/neural_nets/net_builders/categorial.py

```diff
@@ -61,3 +64,60 @@ def build_categoricalmassestimator(
     return CategoricalMassEstimator(
         categorical_net, input_shape=batch_x[0].shape, condition_shape=batch_y[0].shape
     )
+
+
+def build_autoregressive_categoricalmassestimator(
+    batch_x: Tensor,
+    batch_y: Tensor,
+    z_score_x: Optional[str] = "none",
+    z_score_y: Optional[str] = "independent",
+    num_hidden: int = 20,
+    num_layers: int = 2,
+    categories: Optional[Tensor] = None,
```

Comment:
add to docstring. 

I suggest renaming this to `num_categories`. and I like the approach with the warning that if it is `None ` it will be inferred from the data.